app/src/main/java/com/mathewsachin/fategrandautomata/ui/skill_maker/SkillMakerViewModel.kt

```diff
@@ -160,7 +160,7 @@ class SkillMakerViewModel @Inject constructor(
         add(
             SkillMakerEntry.Action(
                 when (skill) {
-                    is Skill.Servant -> AutoSkillAction.ServantSkill(skill, target)
+                    is Skill.Servant -> AutoSkillAction.ServantSkill(skill, if (target == null) null else listOf(target))
```

Comment:
Kotlin offers `listOfNotNull(nullable)`, which either creates an empty list on `null` or creates a read-only list containing the object.